charmhelpers/contrib/hahelpers/apache.py

```diff
@@ -64,6 +70,42 @@ def get_cert(cn=None):
     return (cert, key)
 
 
+def validate_cert(cert, key, ca):
+    """
+    cert (bytes): cert.crt file contents (PEM armored)
+    key (bytes): cert.key file contents
+    ca (Optional[bytes]): None or private ca contents
+
+    return True if ok
+    otherwise return False, set blocked status, and set useful error message
+    """
+
+    context = None
+    if ca:
+        context = ValidationContext(trust_roots=[ca])
```

Comment:
`ca` may be a bundle of CAs. So this example would help:
https://github.com/wbond/certvalidator/blob/master/docs/usage.md#custom-trust-rootsca-certs

Also, there may be a corner case where `ssl_ca` is used for unrelated services but `ssl_cert` can be validated via the system CA chain. So I would prefer using `extra_trust_roots` instead of `trust_roots`, which excludes the system's chain.